File: java/Intervals/MergeOverlappingIntervals.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import DS.Interval;

/*
    // Definition of Interval:
    class Interval {
        public int start;
        public int end;
        public Interval(int start, int end) {
            this.start = start;
            this.end = end;
        }
    }
 */

public class MergeOverlappingIntervals {
    public List<Interval> mergeOverlappingIntervals(List<Interval> intervals) {
        Collections.sort(intervals, (a, b) -> Integer.compare(a.start, b.start));
        List<Interval> merged = new ArrayList<>();
        merged.add(intervals.get(0));
        for (int i = 1; i < intervals.size(); i++) {
            Interval B = intervals.get(i);
            Interval A = merged.get(merged.size() - 1);
            // If A and B don't overlap, add B to the merged list.
            if (A.end < B.start) {
                merged.add(B);
            }
            // If they do overlap, merge A with B.
            else {
                A.end =  Math.max(A.end, B.end);
            }
        }
        return merged;
    }   
}

File: csharp/Intervals/MergeOverlappingIntervals.cs
using ds.Interval;

/*
	Definition of Interval:
	class Interval
    {
        public Interval(int start, int end)
        {
            Start = start; 
            End = end;
        }

        public int Start { get; set; }

        public int End { get; set; }
    }
*/

public class Solution
{
    public Interval[] MergeOverlappingIntervals(Interval[] intervals)
    {
        Array.Sort(intervals, (a, b) => a.Start - b.Start);
        List<Interval> merged = [intervals[0]];

        foreach (Interval B in intervals[1..])
        {
            Interval A = merged[^1];

            // If A and B don't overlap, add B to the merged list.
            if (A.End < B.Start)
                merged.Add(B);
            // If they do overlap, merge A with B.
            else
                merged[^1] = new Interval(A.Start, Math.Max(A.End, B.End));
        }

        return [.. merged];
    }
}
